### PR TITLE
Fix Indirect Data Reduction crash when open/close interface multiple times

### LINKED ISSUE #37409: Crash after using multiple Indirect/Inelastic interfaces
**Describe the bug**
There is a crash after using multiple Indirect/inelastic interfaces.

**To Reproduce**

For example
1. Go to Inelastic Data Manipulation -> Symmetrise
2. Load a _red workspace and click `Run`. Close the GUI
3. Go to Indirect Data Reduction -> ISIS Diagnositics
4. Load 26176 and click `Run`. Close the GUI
5. Go to Indirect Diffraction
6. Select OSIRIS, and load 137793 as sample. 137713 as vanadium. Click Run
7. Crash!

**Expected behavior**
There should be no crash

**Platform/Version (please complete the following information):**
 - OS: Found on Windows

### PR DESCRIPTION
**Pull request by robertapplin**
### Description of work
This PR fixes a crash when you open Indirect Data Reduction, close it, and then open it again. The presenter for the ISIS Energy Transfer tab was not being destructed because the presenter was being held as a raw pointer. This meant that the when a working changes in the ADS, the OutputOptions held by the presenter were trying to update the widget in the previously deleted interface. The solution was to use a unique_ptr to hold the presenters for the Data Reduction interface.

This PR does two additional things:
- It ensures the `OutputOptionsPresenter` uses the ADSObserver class rather than implementing Poco observers itself. This is clearer to read, and ensures the same presenter does not subscribe to a poco observer more than once (this can cause issues as seen previously in #37074
- It ensures the Data Manipulation interface uses unique ptrs to hold their presenters. The crash was not being seen on Data Manipulation because the presenters are QObjects, and get destructed when the parent widget is destroyed

Fixes #37409

### To test:
1. Open Indirect Data Reduction. Close it. Open it again
2. No crash!
3. Also try the instructions in the attached issue

*This does not require release notes* because **I believe this was introduced recently when IETPresenter was made to be qt-independent**

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: qt/scientific_interfaces/Inelastic/Manipulation/DataManipulationInterface.cpp [hunk 1]
```diff
@@ -39,16 +36,16 @@ DECLARE_SUBWINDOW(DataManipulationInterface)
...
 
 std::string DataManipulationInterface::documentationPage() const { return "Inelastic Data Manipulation"; }
 
 /**
  * Called when the user clicks the Python export button.
  */
 void DataManipulationInterface::exportTabPython() {
-  QString tabName = m_uiForm.twIDRTabs->tabText(m_uiForm.twIDRTabs->currentIndex());
-  m_tabs[tabName].second->exportPythonScript();
+  auto const tabName = m_uiForm.twIDRTabs->tabText(m_uiForm.twIDRTabs->currentIndex()).toStdString();
```
Comment (cailafinn) [inline qt/scientific_interfaces/Inelastic/Manipulation/DataManipulationInterface.cpp:47]:
```suggestion
  auto const &tabName = m_uiForm.twIDRTabs->tabText(m_uiForm.twIDRTabs->currentIndex()).toStdString();
```

**Review by cailafinn — CHANGES_REQUESTED** [review]
Code looking good, just noticed a few places to make use of ref assignment. I'm aware much of this is due for changes when converting to MVP, so I'm happy for that to be maintenance if you think it's too much of a pain to re-run the CI. 

Otherwise crashes are resolved, tried many different variations of opening and closing windows, while running etc. and had no crashes. 

Just want to check if the crash behaviour is unit-testable?

**Comment by robertapplin** [comment]
> 
> Just want to check if the crash behaviour is unit-testable?

Thanks for the review! Unfortuantely this area is not easily unit testable - I'm hoping this will change in the future when MVP has been completed

**Review by cailafinn — APPROVED** [review]
Amendments to the code look good. Still working as expected without crashing when opening/closing GUIs. Changes to docs unneeded. Not currently unit-testable. Nice work.

**Pull request merged**